editor/src/clj/editor/fxui.clj

```diff
@@ -549,40 +567,43 @@
                                        :error "text-area-error"
                                        :borderless "text-area-borderless"))))
 
-(def ^:private ext-with-popup-on-props
+(def ^:private ext-with-popup-window-on-props
   (fx/make-ext-with-props
     {:on (fx.prop/make
            (fx.mutator/adder-remover
-             (fn [^Popup popup [^Node on anchor-x anchor-y]]
+             (fn [^PopupWindow popup [^Node on anchor-x anchor-y on-window]]
                (condp instance? on
-                 Node (.show popup ^Node on (double anchor-x) (double anchor-y))
+                 Node (if on-window
+                        (.show popup (.getWindow (.getScene ^Node on)) (double anchor-x) (double anchor-y))
+                        (.show popup ^Node on (double anchor-x) (double anchor-y)))
                  Window (.show popup ^Window on (double anchor-x) (double anchor-y))))
-             (fn [^Popup popup _]
+             (fn [^PopupWindow popup _]
                (.hide popup)))
-           (tuple-lifecycle fx.lifecycle/dynamic fx.lifecycle/scalar fx.lifecycle/scalar))}))
+           (tuple-lifecycle fx.lifecycle/dynamic fx.lifecycle/scalar fx.lifecycle/scalar fx.lifecycle/scalar))}))
 
-(defn with-popup
-  "Helper popup lifecycle that adds a managed popup to :desc node
+(defn with-popup-window
+  "Helper lifecycle that adds a managed popup window to the wrapped node
 
   Supported props:
-    :desc        node or window desc that will show a popup
-    :showing     whether the popup is showing
-    :anchor-x    screen anchor x
-    :anchor-y    screen anchor y
-    ...the rest of :popup props"
-  [{:keys [desc showing anchor-x anchor-y]
-    :or {anchor-x 0.0
-         anchor-y 0.0}
-    :as props}]
+    :desc    required node or window desc that will show a popup
+    :popup   optional popup window desc, with these additional props:
```

Comment:
I also made a popup optional, so that it's easier in the user code to configure the popup when it's hidden.